### PR TITLE
test: cover the decks service layer

### LINKED ISSUE #5383: Cover the decks service layer
**File:** `rslib/src/decks/service.rs` (320 lines, 0%) · **Target: ≥ 70%**

Protobuf entry point for all deck operations. No math — routing and persistence.

Scope:

- create deck → retrievable by name and by id;
- rename deck persists;
- delete deck → leaves the list; behavior for cards in the removed deck;
- move cards between decks (`set_deck`/equivalent) reassigns correctly;
- proto↔domain (`DeckId`) round-trips.

### PR DESCRIPTION
**Pull request by fernandolins**
## Linked issue

Closes #5383

## Summary / motivation

`rslib/src/decks/service.rs`: the Protobuf entry point for all deck operations, it had 0% test coverage. This PR adds a golden `#[cfg(test)]` suite that exercises the happy path, error paths, boundary cases, and key invariants of the `DecksService` layer and its proto↔domain conversions.

No production code was changed: the module is a thin routing/conversion layer with no logic that warranted refactoring. Testing at this boundary is additive rather than duplicative: `reparent_decks`, `remove_decks_and_child_decks` and the proto↔domain conversions had no direct unit tests before.

Coverage of `decks/service.rs` goes from **0% → ~99.7% of lines**

### Details

- `just check` passes (fmt, clippy, rust_test, and the rest of the suite).
- Run just this module: `cargo test -p anki decks::service::tests`.
- Coverage inspected with `just test-rust --coverage --html`
- `decks/service.rs` reports ~99.7% line coverage.

Scenarios covered:
- proto↔domain conversions: full-field `Deck` round trip for both kinds, `missing kind → InvalidInput`, `FilteredDeckForUpdate` field preservation and `config: None → default`.
- create/retrieve, update (common + kind-specific persistence), rename with descendant cascade, delete + card removal, reparent (incl. `new_parent == 0`).
- name/tree listing: `include_filtered`, empty vs non-empty default skipping, child-name filtering, `deck_tree` counts by `now`, legacy tree JSON.
- current deck, collapsed-scope isolation, legacy JSON paths (incl. `preserve_usn_and_mtime`), and filtered-deck build.
- Error paths assert the specific `AnkiError` variant (`NotFound` / `InvalidInput`), not just `is_err()`.

### CODE REVIEW

File: rslib/src/decks/service.rs [hunk 1]
```diff
@@ -318,3 +316,747 @@ fn deck_names_to_proto(names: Vec<(DeckId, String)>) -> anki_proto::decks::DeckN
...
+        assert_eq!(value["mod"], 123);
+        assert_eq!(value["usn"], 456);
+    }
+
+    // ----------------------------------------------------------------------
+    // Filtered decks
+    // ----------------------------------------------------------------------
+
+    #[test]
+    fn filtered_deck_order_labels_match_the_order_enum() {
```
Comment (abdnh) [inline rslib/src/decks/service.rs:985]: This test duplicates search_order_label()'s implementation details. I wonder if there's a better way to test this to make the test useful without that, maybe checking enum count, non-empty and unique labels?

File: rslib/src/decks/service.rs [hunk 2]
```diff
@@ -318,3 +316,747 @@ fn deck_names_to_proto(names: Vec<(DeckId, String)>) -> anki_proto::decks::DeckN
...
+    use super::*;
+    use crate::decks::FilteredDeck;
+    use crate::decks::FilteredSearchOrder;
+    use crate::decks::FilteredSearchTerm;
+    use crate::prelude::*;
+    use crate::services::DecksService;
+    use crate::tests::DeckAdder;
+    use crate::tests::NoteAdder;
+
+    fn deck_id(did: i64) -> anki_proto::decks::DeckId {
```
Comment (abdnh) [inline rslib/src/decks/service.rs:331]: Nitpick: We can replace all calls of this with just `deck.id.into()` to exercise the `From<DeckId>` implementation at line 240?

**Review by abdnh — APPROVED** [review]
